src/python/grpcio/grpc/_cython/_cygrpc/aio/server.pyx.pxi

```diff
@@ -188,13 +195,19 @@ async def _handle_unary_stream_rpc(object method_handler,
     await execute_batch(rpc_state, ops, loop)
 
 
-async def _handle_cancellation_from_core(object rpc_task,
-                                         RPCState rpc_state,
-                                         object loop):
+async def _schedule_rpc_coro_and_handle_cancellation(object rpc_coro,
```

Comment:
Whenever a function has "and" in the name, it's a good idea to think about breaking it into two separate functions with orthogonal responsibilities.

Reply:
Split into `_schedule_rpc_coro` and `_handle_cancellation_from_core`, and the former function is calling the latter function. The relationship between this two functions shouldn't be parallel, otherwise the race condition still exists.